packages/agents-manager/src/components/agent-dock/agent.tsx

```diff
@@ -0,0 +1,220 @@
+/**
+ * Agent Component
+ *
+ * Handles the chat interface, message loading, and conversation management.
+ */
+
+import {
+	createOdieBotId,
+	getAgentManager,
+	type Message,
+	useAgentChat,
+	type UseAgentChatConfig,
+} from '@automattic/agenttic-client';
+import { AgentUI, EmptyView, type Suggestion } from '@automattic/agenttic-ui';
+import { AgentsManagerSelect } from '@automattic/data-stores';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useCallback, useEffect, useRef } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { useNavigate, useParams } from 'react-router-dom';
+import { API_BASE_URL } from '../../constants';
+import useLoadConversation from '../../hooks/use-load-conversation';
+import { AGENTS_MANAGER_STORE } from '../../stores';
+import { lastConversationCache } from '../../utils/conversation-cache';
+import BigSkyIcon from '../big-sky-icon';
+import ChatHeader, { type Options as ChatHeaderOptions } from '../chat-header';
+import ChatMessageSkeleton from '../chat-message-skeleton';
+import { AI } from '../icons';
+import type { ComponentType } from 'react';
+
+interface AgentProps {
+	agentConfig: UseAgentChatConfig;
+	emptyViewSuggestions: Suggestion[];
+	getChatHeaderOptions: () => ChatHeaderOptions;
+	isDocked: boolean;
+	messageRenderer: ComponentType< { children: string } >;
+	onClose: () => void;
+}
+
+export default function Agent( {
+	agentConfig,
+	emptyViewSuggestions,
+	getChatHeaderOptions,
+	isDocked,
+	messageRenderer,
+	onClose,
+}: AgentProps ) {
+	const { chatId = '' } = useParams< { chatId?: string } >();
```

Comment:
We should use `sessionId` instead of `chatId` for consistency. A session may include multiple chats.